Ncsdk 19394 event manager section placement (#10093)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

  **Workaround:** Manually cherry-pick and apply the commit with the fix from the ``main`` branch (commit hash: ``7e8c23a6632632f0cee885abe955e37a6942911d``).

> .. rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0

Author: No. It would not. After the change how subscribers are processed they all goes into single big sections and there some special, zero size, tags are used to mark begin and end of the specific event subscribers. This tags are placed here thanks to name sorting and this is sensitive to the right name usage.

In old implementation every event listeners group, on every level, goes into its specyfic section. So the section named super_event_prioX is not the same like super_event_prioX_prioX. It means that in old implementation the issue should not appear.

> .. rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0

Reviewer: I missed the fact that section name used to contain the `prioX`. Please disregard the comment.

> NCSIDB-925: Event listeners in :ref:`app_event_manager` may overlap when using non default naming convention.

Reviewer: subscribers?

Author: Fixed.

>   Due to relaying on section sorting by the name and added "magic" postfixes, to locate the event listener start and stop address ranges,
>   using non default event naming scheme may break the expected listeners sorting.

Reviewer 2: ```suggestion
In order to locate the event listener's start and stop address ranges, sections have to be sorted by the name with added postfixes.
Using a non-default event naming scheme may break the expected listeners sorting.
```

>   using non default event naming scheme may break the expected listeners sorting.

Author: I think it was already fixed by @annwoj proposals.

>   In such situation one of the event listeners array may be actually placed inside the other.

Reviewer 2: ```suggestion
  In this situation, one of the event listeners arrays might be placed inside the other.
```

>   This results in the execution of the event listeners also when the event that contains its listeners inside is processed.

Reviewer 2: ```suggestion
  Because of this, event listeners are executed even when they are placed inside the event. 
```

Reviewer 2: please double-check if the meaning is correct

Author: The option created by @divipillai chosen.

>   This results in the execution of the event listeners also when the event that contains its listeners inside is processed.

Reviewer 2: another version of the above sentence, @rakons please select the one that reflect the meaning better :)

>   New implementation with the usage of section naming that is imposible to be used as event name (not valid variable identifier) resolved this issue.

Reviewer 2: ```suggestion
  To resolve this issue, a new implementation was introduced, that disables using event names as section names (invalid variable identifier).
```

> NCSIDB-925: Event listeners in the :ref:`app_event_manager` may overlap when using a non-default naming convention
> In order to locate the event listener's start and stop address ranges, sections have to be sorted by the name with added postfixes.
> Hence, using a non-default event naming scheme may break the expected listeners sorting.
>   In this situation, one of the event listeners arrays might be placed inside the other.
>   So, when the event related to the outer listeners is processed, the event listeners that are inside are also executed.
>   To resolve this issue, a new implementation was introduced that uses a section naming that cannot be used as event name (invalid variable identifier).

Reviewer 3: ```suggestion
NCSIDB-925: Event listeners in the :ref:`app_event_manager` may overlap when using a non-default naming convention
  In order to locate the event listener's start and stop address ranges, sections have to be sorted by the name with added postfixes.
  Hence, using a non-default event naming scheme may break the expected listeners sorting.
  In this situation, one of the event listeners arrays might be placed inside the other.
  So, when the event related to the outer listeners is processed, the event listeners that are inside are also executed.
  To resolve this issue, a new implementation was introduced that uses a section naming that cannot be used as event name (invalid variable identifier).
```
You may get a doc build failure.

Author: Yes. Thanks. I had already located this issue during manual implementation of @MarekPieta comments. Already fixed (locally).

>   **Workaround:** Stick to the default event names where each event ends with _event postfix.

Reviewer 2: ```suggestion
  **Workaround:** Use the default event names, ensuring that each event name ends with the relevant postfix.
  Make sure that the event name does not start the same way as another event. For example, creating the following events: RX_EVENT and RX_EVENT_ERROR_EVENT would still cause the issue.
```

>   It would guarantee proper listeners sections sortng.

Reviewer 2: ```suggestion
```

>   It would guarantee proper listeners sections sortng.

Author: Already fixed by @annwoj proposal.

  **Workaround:** Manually cherry-pick and apply the commit with the fix from the ``main`` branch (commit hash: ``7e8c23a6632632f0cee885abe955e37a6942911d``).

.. _ncsidb_925:

.. rst-class:: v2-2-0 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0

NCSIDB-925: Event subscribers in the :ref:`app_event_manager` may overlap when using a non-default naming convention
  In order to locate the event subscribers start and stop address ranges, sections have to be sorted by the name with added postfixes.
  Hence, using a non-default event naming scheme may break the expected subscribers sorting.
  In this situation, one of the event subscribers arrays might be placed inside the other.
  So, when the event related to the outer subscribers is processed, the event subscribers that are inside are also executed.
  To resolve this issue, a new implementation was introduced that uses a section naming that cannot be used as event name (invalid variable identifier).

  **Workaround:** Use the default event names, ensuring that each event name ends with the ``_event`` postfix.

>   Make sure that the event name does not start the same way as another event. For example, creating the following events: ``rx_event`` and ``rx_event_error_event`` would still cause the issue.

Reviewer: One sentence per line

Reviewer: `would cause the issue`.
I think we could omit word `still`, because it may lead to confusion (the issue is already fixed so that the issue would no longer replicate).

Reviewer: btw. Do we still need the first sentence of the described workaround?

Author: Multiple sentences in one line fixed.
